DB/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -46,11 +143,65 @@ public void close() {
     public void refresh() {
         streamPublisher.flush();
         // Switch columns, update index, deliver notification
+
+        final long oldSize = index.size();
+        final long newSize;
+
+        synchronized (this) {
```

Comment:
We only need to hold the lock while swapping the buffered sources. Meaning we should clear out the prevColumnSources `(if oldSize != 0)`, then grab our lock, then check buffered size, then do the swap under the lock, then release the lock and do the rest of our maintenance.